cpp/test/distance/gram.cu

```diff
@@ -127,6 +129,9 @@ class GramMatrixTest : public ::testing::TestWithParam<GramMatrixInputs> {
 
     (*kernel)(handle, x1_span, x2_span, out_span);
 
+    // Something in gram is executing not on the 'stream' and therefore
+    // a full device sync is required
+    RAFT_CUDA_TRY(cudaDeviceSynchronize());
```

Comment:
Can we open an issue to track this problem?

CC @cjnolet

Reply:
I have updated the test to remove the usage of multiple streams which caused the sync issues.

Reply:
Please synchronize the stream in the handle or use interruptible to sync the stream instead of calling cuda APIs directly

Reply:
PR has been updated to fix the underlying bug that caused the `cudaDeviceSynchronize` to be needed.